### DISCUSSION TITLE
Map a key to toggle dynamic resize?

### DISCUSSION THREAD

**Discussion by brucejxz** [nvim-tree/nvim-tree.lua#2661] [Q&A] [answered]
Is it possible to map a key that will toggle the dynamic width functionality that you get by setting:

```
view = {
  width = {}
}
```

The idea is that I would like a narrow tree most of the time and don't care if some of the file names are too long. However, sometimes I want to see everything, and when I do, I want to toggle the width to be exactly wide enough to fit everything.

**Comment by alex-courtis** [comment, marked as answer]
That is indeed functionality that I am interested in. Try this out. We could add a recipe or some other support.

```lua
local view_width_max = -1

local function toggle_width_unbounded()
  if view_width_max == -1 then
    view_width_max = 30
  else
    view_width_max = -1
  end
  require("nvim-tree.api").tree.reload()
end

local function get_view_width_max()
  return view_width_max
end

vim.keymap.set("n", "<space>f", toggle_width_unbounded, { noremap = true })

require("nvim-tree").setup({
  view = {
    width = {
      min = 30,
      max = get_view_width_max,
    }
  }
})
```

**Reply by brucejxz** [reply]
Thanks a lot! Yeah, that does the trick. And for anyone else, if you want the tree to start out "minimised" when you first open it, change:

```
local view_width_max = -1
```
to:
```
local view_width_max = 30
```

**Reply by alex-courtis** [reply]
https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes#toggle-adaptive-width
